daemon-scheduler/pkg/api/v1/api_test.go

```diff
@@ -64,13 +70,119 @@ func (suite *APITestSuite) TestPing() {
+}
+
+func (suite *APITestSuite) TestListEnvironments() {
+	e1, err := types.NewEnvironment("e1", "td1", "cluster1")
+	e2, err := types.NewEnvironment("e2", "td2", "cluster2")
+	environments := []types.Environment{*e1, *e2}
+	suite.environment.EXPECT().ListEnvironments(gomock.Any()).Return(environments, nil)
+	request, err := http.NewRequest("GET", "/v1/environments", nil)
+	assert.Nil(suite.T(), err, "Unexpected error")
+
+	responseRecorder := httptest.NewRecorder()
+	suite.router.ServeHTTP(responseRecorder, request)
+
+	assert.Equal(suite.T(), http.StatusOK, responseRecorder.Code)
 
-	s.Path("/ping").
-		Methods("GET").
-		HandlerFunc(suite.api.Ping)
+	var environmentsModel models.Environments
+	b, _ := ioutil.ReadAll(responseRecorder.Body)
+	json.Unmarshal(b, &environmentsModel)
 
-	return s
+	for i := 0; i < len(environments); i++ {
+		environment := environments[i]
+		environmentModel := environmentsModel.Items[i]
+		suite.assertSame(&environment, environmentModel)
+	}
+}
+
+func (suite *APITestSuite) TestListEnvironmentsServerError() {
+	err := errors.New("Error when calling ListEnvironments")
+	suite.environment.EXPECT().ListEnvironments(gomock.Any()).Return(nil, err)
+	request, err := http.NewRequest("GET", "/v1/environments", nil)
+	assert.Nil(suite.T(), err, "Unexpected error")
+
+	responseRecorder := httptest.NewRecorder()
+	suite.router.ServeHTTP(responseRecorder, request)
+
+	assert.Equal(suite.T(), http.StatusInternalServerError, responseRecorder.Code)
+}
+
+func (suite *APITestSuite) TestDeleteEnvironment() {
+	name := "testEnv"
+	suite.environment.EXPECT().DeleteEnvironment(gomock.Any(), name).Return(nil)
+
+	request, err := http.NewRequest("DELETE", "/v1/environments/"+name, nil)
+	assert.Nil(suite.T(), err, "Unexpected error")
+
+	responseRecorder := httptest.NewRecorder()
+	suite.router.ServeHTTP(responseRecorder, request)
+
+	assert.Equal(suite.T(), http.StatusOK, responseRecorder.Code)
+}
+
+func (suite *APITestSuite) TestDeleteEnvironmentMissingEnvironment() {
+	name := "testEnv"
+	notfounderr := types.NewNotFoundError(errors.New("Environment is missing"))
+	suite.environment.EXPECT().DeleteEnvironment(gomock.Any(), name).Return(notfounderr)
+
+	request, err := http.NewRequest("DELETE", "/v1/environments/"+name, nil)
+	assert.Nil(suite.T(), err, "Unexpected error")
+
+	responseRecorder := httptest.NewRecorder()
+	suite.router.ServeHTTP(responseRecorder, request)
+
+	assert.Equal(suite.T(), http.StatusNotFound, responseRecorder.Code)
+}
+
+func (suite *APITestSuite) assertSame(environment *types.Environment, environmentModel *models.Environment) {
+	assert.Equal(suite.T(), environment.Name, aws.StringValue(environmentModel.Name))
+	assert.Equal(suite.T(), environment.Cluster, environmentModel.InstanceGroup.Cluster)
+	assert.Equal(suite.T(), environment.DesiredTaskDefinition, environmentModel.TaskDefinition)
+}
+
+func (suite *APITestSuite) getRouter() *mux.Router {
+	return NewRouter(suite.api)
```

Comment:
I get that this actually tests the router and also prevents us from re-writing the router. But the reason I had re-written it was so we explicitly define invalid paths in the router to make sure that the handler functions handle them in case the call reaches it (Ex. https://github.com/blox/blox/blob/master/cluster-state-service/handler/api/v1/taskapis_test.go#L399). Might be an overkill. Your call!